modules/afmongodb/afmongodb.c

```diff
@@ -370,83 +394,128 @@ afmongodb_worker_insert (LogThrDestDriver *s, LogMessage *msg)
+                    evt_tag_str("driver", self->super.super.super.id));
+          return WORKER_INSERT_RESULT_ERROR;
         }
     }
 
-  if (!success && (errno == ENOTCONN))
-    return WORKER_INSERT_RESULT_NOT_CONNECTED;
+  return WORKER_INSERT_RESULT_SUCCESS;
+}
 
-  if (!success)
-    return WORKER_INSERT_RESULT_ERROR;
+gboolean
+afmongodb_dd_private_uri_init(LogDriver *d)
+{
+  MongoDBDestDriver *self = (MongoDBDestDriver *)d;
 
-  return WORKER_INSERT_RESULT_SUCCESS;
+#if SYSLOG_NG_ENABLE_LEGACY_MONGODB_OPTIONS
+  if (!afmongodb_dd_create_uri_from_legacy(self))
+    return FALSE;
+#endif
+
+  if (!self->uri_str)
+    self->uri_str = g_string_new(DEFAULT_URI);
+
+  self->uri_obj = mongoc_uri_new(self->uri_str->str);
+  if (!self->uri_obj)
+    {
+      msg_error("Error parsing MongoDB URI",
+                evt_tag_str("uri", self->uri_str->str),
+                evt_tag_str("driver", self->super.super.super.id));
+      return FALSE;
+    }
+
+  self->const_db = mongoc_uri_get_database(self->uri_obj);
+  if (!self->const_db || !strlen(self->const_db))
+    {
+      msg_error("Missing DB name from MongoDB URI",
+                evt_tag_str("uri", self->uri_str->str),
+                evt_tag_str("driver", self->super.super.super.id));
+      return FALSE;
+    }
+
+  msg_verbose("Initializing MongoDB destination",
+              evt_tag_str("uri", self->uri_str->str),
+              evt_tag_str("db", self->const_db),
+              evt_tag_str("collection", self->coll),
+              evt_tag_str("driver", self->super.super.super.id));
+
+  return TRUE;
 }
 
 static void
-afmongodb_worker_thread_init(LogThrDestDriver *d)
+_worker_thread_init(LogThrDestDriver *d)
 {
   MongoDBDestDriver *self = (MongoDBDestDriver *)d;
 
-  afmongodb_dd_connect(self, FALSE);
+  _dd_connect(self, FALSE);
 
   self->current_value = g_string_sized_new(256);
 
   self->bson = bson_sized_new(4096);
 }
 
 static void
-afmongodb_worker_thread_deinit(LogThrDestDriver *d)
+_worker_thread_deinit(LogThrDestDriver *d)
 {
   MongoDBDestDriver *self = (MongoDBDestDriver *)d;
 
-  g_string_free (self->current_value, TRUE);
+  if (self->current_value)
+    {
+      g_string_free(self->current_value, TRUE);
+      self->current_value = NULL;
+    }
 
-  bson_free (self->bson);
+  bson_destroy(self->bson);
+  self->bson = NULL;
 }
 
 /*
  * Main thread
  */
 
 static void
-afmongodb_dd_init_value_pairs_dot_to_underscore_transformation(MongoDBDestDriver *self)
+_init_value_pairs_dot_to_underscore_transformation(MongoDBDestDriver *self)
 {
   ValuePairsTransformSet *vpts;
 
- /* Always replace a leading dot with an underscore. */
+  /* Always replace a leading dot with an underscore. */
   vpts = value_pairs_transform_set_new(".*");
   value_pairs_transform_set_add_func(vpts, value_pairs_new_transform_replace_prefix(".", "_"));
   value_pairs_add_transforms(self->vp, vpts);
 }
 
 static gboolean
-afmongodb_dd_init(LogPipe *s)
+_logpipe_init(LogPipe *s)
 {
```

Comment:
I wouldn't include class names in virtual methods like logpipe in this case. I understand that init without a qualifier might be too general as a term though. 

In other destinations we either used the unqualified _init as a name or we kept the original fully qualified name. I think both are OK, but I wouldn't start a new convention as you did here